src/resolve/resolved-etc-hosts.c

```diff
@@ -444,5 +444,5 @@ int manager_etc_hosts_lookup(Manager *m, DnsQuestion* q, DnsAnswer **answer) {
                         return r;
         }
 
-        return 1;
+        return dns_answer_isempty(*answer) ? 0 : 1;
```

Comment:
this change isn't right. Note that hostname resolution generally distuingishes NODATA from NXDOMAIN. Translated to this case this means: If you try to resolve a name, and the name in general exists, but not the RR type/class you were looking for, then you return a successful but empty response — which is the case that is usually called NODATA. However NXDOMAIN is the other case: where there whole domain doesn't exist, and hence neither RRs of teh requested type/class nor of any other name.

hence, we should return 1 here in this case, as we already bailed out much earlier with 0 if we didn't find anything at ll.